ivy_tests/test_ivy/test_frontends/test_numpy/test_indexing_routines/test_indexing_like_operations.py

```diff
@@ -0,0 +1,76 @@
+# Testing Function
+# global
+from hypothesis import strategies as st
+
+# local
+import ivy_tests.test_ivy.helpers as helpers
+
+# from ivy_tests.test_ivy.test_functional.test_core.test_statistical import (
+#   statistical_dtype_values,
+# )
+import ivy_tests.test_ivy.test_frontends.test_numpy.helpers as np_frontend_helpers
+from ivy_tests.test_ivy.helpers import handle_frontend_test
+
+# import ivy
+
+
+@handle_frontend_test(
+    fn_tree="numpy.diagonal",
+    dtype_and_x=helpers.dtype_and_values(
+        available_dtypes=helpers.get_dtypes("numeric"),
+        num_arrays=1,
+        # min_value=0,
+        # max_value = 1,
+        shared_dtype=True,
+        min_num_dims=2,
+    ),
+    where=np_frontend_helpers.where(),
+    # dtype_and_axis1=helpers.dtype_values_axis(
+    #   available_dtypes=helpers.get_dtypes("numeric"),
+    # num_arrays=1,
+    # min_value=0,
+    # max_value = 1,
+    #  shared_dtype=True,
+    # min_num_dims=2,
+    # ),
+    offset=st.integers(min_value=0),  # ,max_value=10),
+    axis1=st.integers(min_value=-2, max_value=2),
+    axis2=st.integers(min_value=-2, max_value=2),
+    # dtype=helpers.get_dtypes("float", full=False, none=True),
+    # where=np_frontend_helpers.where(),
+)
+def test_numpy_diagonal(
+    dtype_and_x,
+    as_variable,
+    num_positional_args,
+    native_array,
+    where,
+    offset,
+    axis1,
+    axis2,
+    with_out,
```

Comment:
You can remove `with_out` from the arguments. We are not using the passed argument in this case, as we are hard-coding `with_out=False` below.

Reply:
if i omit  ` with_out `    then
`E           TypeError: test_frontend_function() missing 1 required keyword-only argument: 'with_out' `
also on `where` 
`E       TypeError: diagonal() got an unexpected keyword argument 'where' `